## ---------------------------
##
## Script name: FitToSimulatedData
##
## Purpose of script: It fits data to simulated data (generated by
##      https://github.com/afodor/metagenomicsTools/blob/master/src/binomFit/HowManyVariants.java)
##
## Author: Farnaz Fouladi
##
## Date Created: 2021-01-28
##
## ---------------------------


#Libraries
library(dplyr)

myT <- read.table("data/china/OneMismatchCluster.txt",sep="\t", header=TRUE)
simulatedData <- read.table("data/china/SimulatedData/depth.txt", sep="\t", header=TRUE)

parents<-myT %>% filter(!is.na(Child)) %>% distinct(Parent,.keep_all=TRUE)

pdf("figures/FitToSimulatedData.pdf",width = 5,height = 5)
plot(log10(parents$ParentAbundance), parents$NumberOfChildren/750,
     ylab="Fraction of all possible one-mismatch variants",
     xlab="Sequencing depth",
     main="Error rate = 0.00015")

N <- seq(min(myT$ParentAbundance),max(myT$ParentAbundance),100)

errorProb <- 0.00015

#Probabilities of observing one-mismatch for each sequencing depth:

probOfSeeingVariant<-sapply(N,function(x) 1-dpois( 0, lambda = x*errorProb/3))
lines(log10(N), probOfSeeingVariant,lty=1,col="red",lwd=2)
lines(log10(simulatedData$depth), simulatedData$numMutations/750,lty=1,col="blue",lwd=2)
legend("topleft",legend = c("Simulated data","Poisson model","Real data"),
       col=c("blue","red","black"),pch=1,cex=0.8)

dev.off()

